core/src/main/java/io/questdb/griffin/SqlOptimiser.java

```diff
@@ -1163,10 +1163,12 @@ private int getIndexOfTableForColumn(QueryModel model, CharSequence column, int
             return orderByAdvice;
         }
 
-        CharSequenceObjHashMap<CharSequence> map = model.getAliasToColumnNameMap();
+        CharSequenceObjHashMap<QueryColumn> map = model.getAliasToColumnMap();
         for (int i = 0; i < len; i++) {
-            orderByAdvice.add(nextLiteral(map.get(orderBy.getQuick(i).token)));
-
+            QueryColumn queryColumn = map.get(orderBy.getQuick(i).token);
```

Comment:
We need more tests.

What happens here is what I've written about in email - if order by has non-literal columns, advice must be empty. We must not give selective advice to the sub=query